### PR TITLE
WebsiteDocs-4: Added the English `code-conduct.md` which was present …

### PR DESCRIPTION
**Pull request by IamMujuziMoses**
…in Chinese but wasn't present in English and how to access it in 'development.js' file and how to access the Chinese version too

### CODE REVIEW

**Comment by IamMujuziMoses** [comment]
This PR is associated to this main [issue]

[issue]: https://github.com/apache/incubator-dolphinscheduler/issues/3562

**Comment by CalvinKirs** [comment]
@Segun-Ogundipe @Tianqi-Dotes PTAL.Thx

File: docs/en-us/development/code-conduct.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+## Code of Conduct
+
+The following Code of Conduct is based on full compliance with the [Apache Software Foundation Code of Conduct](https://www.apache.org/foundation/policies/conduct.html).
+
+### Development philosophy.
+ - **Consistent** code style, naming, and usage are consistent.  
+ - **Easy to read** code is ambiguous, easy to read and understand, when debugging one knows the intent of the code.
```
Comment (Tianqi-Dotes) [inline docs/en-us/development/code-conduct.md:7]: the word ambiguous
differ from the original doc
maybe use clear nonambiguous obvious plain unambiguous unequivocal accurate instead

File: docs/en-us/development/code-conduct.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ - **Neat** agree with the concepts of《Refactoring》and《Code Cleanliness》and pursue clean and elegant code.
+ - **Abstract** hierarchy is clear and the concepts are refined and reasonable. Keep methods, classes, packages, and modules at the same level of abstraction.
+ - **Heart** Maintain a sense of responsibility and continue to be carved in the spirit of artisans.
+ 
+### Development specifications.
+
+ - Executing `mvn -U clean package -Prelease` can compile and test through all test cases. 
+ - The test coverage tool checks for no less than dev branch coverage.
+ - Use Checkstyle to check your code for special reasons for violating validation rules. The template location is located at ds_check_style.xml.
```
Comment (Tianqi-Dotes) [inline docs/en-us/development/code-conduct.md:16]: didn't mention this file is under the root directory

File: docs/en-us/development/code-conduct.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+### Coding specifications.
+
+ - Use linux line breaks.
+ - Indentation (including empty lines) is consistent with the last line.
+ - An empty line is required between the class declaration and the following variable or method.
+ - There should be no meaningless empty lines.
+ - Classes, methods, and variables should be named as the name implies and abbreviations should be avoided.
+ - Return value variables are named after `result`; `each` is used in loops to name loop variables; and `entry` is used in map instead of `each`.
+ - The cached exception is called `e`; Catch the exception and do nothing, and the exception is named `ignored`.
+ - Profiles are named in camelCase, and file names are lowercase.
```
Comment (Tianqi-Dotes) [inline docs/en-us/development/code-conduct.md:28]: ambiguous meaning in 2 parts.
the word 'Profiles'  maybe change to the 'configuration file' would be more accurate
'and file names are lowercase.' didn't mention the initial letter should be lowercase.

File: docs/en-us/development/code-conduct.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+ - `equals` and `==` In a conditional expression, the constant is left, the variable is on the right, and in the expression greater than less than condition, the variable is left and the constant is right.
+ - In addition to the abstract classes used for inheritance, try to design the class as `final`.
+ - Nested loops are as much a method as possible.
+ - The order in which member variables are defined and the order in which parameters are passed is consistent across classes and methods.
+ - Priority is given to the use of guard statements.
+ - Classes and methods have minimal access control.
+ - The private method used by the method should follow the method, and if there are multiple private methods, the writing private method should appear in the same order as the private method in the original method.
+ - Method entry and return values are not allowed to be `null`.
+ - The return and assignment statements of if else are preferred with the tri-objective operator.
+ - Priority is given to `LinkedList` and only use `ArrayList` if you need to get element values in the collection.
```
Comment (Tianqi-Dotes) [inline docs/en-us/development/code-conduct.md:39]: didn't mention get element values through index when using ArrayList.

File: docs/en-us/development/code-conduct.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+ - Collection types such as `ArrayList`，`HashMap` that may produce expansion must specify the initial size of the collection to avoid expansion.
+ - Logs and notes are always in English.
+ - Comments can only contain `javadoc`, `todo` and `fixme`.
+ - Exposed classes and methods must have javadoc, other classes and methods and methods that override the parent class do not require javadoc.
+
+### Unit test specifications.
+
+ - Test code and production code are subject to the same code specifications.
+ - Unit tests are subject to AIR (Automatic, Independent, Repeatable) Design concept.
+   - Automatic : Unit tests should be fully automated, not interactive. Manual checking of output results is prohibited, `System.out`, `log`, etc. are not allowed, and must be verified with assertions.
```
Comment (Tianqi-Dotes) [inline docs/en-us/development/code-conduct.md:49]: try to fix space after the  word Automatic

File: docs/en-us/development/code-conduct.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+ - Logs and notes are always in English.
+ - Comments can only contain `javadoc`, `todo` and `fixme`.
+ - Exposed classes and methods must have javadoc, other classes and methods and methods that override the parent class do not require javadoc.
+
+### Unit test specifications.
+
+ - Test code and production code are subject to the same code specifications.
+ - Unit tests are subject to AIR (Automatic, Independent, Repeatable) Design concept.
+   - Automatic : Unit tests should be fully automated, not interactive. Manual checking of output results is prohibited, `System.out`, `log`, etc. are not allowed, and must be verified with assertions. 
+   - Independent : It is prohibited to call each other between unit test cases and to rely on the order of execution. Each unit test can be run independently.
```
Comment (Tianqi-Dotes) [inline docs/en-us/development/code-conduct.md:50]: try to fix space after the  word Independent

File: docs/en-us/development/code-conduct.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+   - Repeatable : Unit tests cannot be affected by the external environment and can be repeated.
```
Comment (Tianqi-Dotes) [inline docs/en-us/development/code-conduct.md:51]: try to fix space after the word Repeatable

File: docs/en-us/development/code-conduct.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+### Unit test specifications.
+
+ - Test code and production code are subject to the same code specifications.
+ - Unit tests are subject to AIR (Automatic, Independent, Repeatable) Design concept.
+   - Automatic : Unit tests should be fully automated, not interactive. Manual checking of output results is prohibited, `System.out`, `log`, etc. are not allowed, and must be verified with assertions. 
+   - Independent : It is prohibited to call each other between unit test cases and to rely on the order of execution. Each unit test can be run independently.
+   - Repeatable : Unit tests cannot be affected by the external environment and can be repeated. 
+ - Unit tests are subject to BCDE（Border, Correct, Design, Error) Design principles.
+   - Border (Boundary value test) : The expected results are obtained by entering the boundaries of loop boundaries, special values, data order, etc.
```
Comment (Tianqi-Dotes) [inline docs/en-us/development/code-conduct.md:53]: try to fix space after the )

File: docs/en-us/development/code-conduct.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+ - Prepare the environment for code separation from the test code.
+ - Only jUnit `Assert`，hamcrest `CoreMatchers`，Mockito Correlation can use static import.
+ - Single-data assertions should use `assertTrue`，`assertFalse`，`assertNull` and `assertNotNull`.
+ - Multi-data assertions should use `assertThat`.
+ - Accurate assertion, try not to use `not`，`containsString` assertion.
+ - The true value of the test case should be named actualXXX, and the expected value should be named expectedXXX.
+ - Classes and Methods with `@Test` labels do not require javadoc.
+
+ - Public specifications.
+   - Each line is no longer than `200` in length, ensuring that each line is semanticly complete for easy understanding.
```
Comment (Tianqi-Dotes) [inline docs/en-us/development/code-conduct.md:68]: wrong spelling word semanticly

**Review by Tianqi-Dotes — CHANGES_REQUESTED** [review]
hi MujuziMoses,
I have reviewed the code-conduct.md doc, it is nice and I found some places not accurate from the original. please check those places.
Thanx
sincerely
Tianqi

**Comment by IamMujuziMoses** [comment]
@Tianqi-Dotes sure thing, thanks for the review, let me get working on those

**Comment by CalvinKirs** [comment]
Hi, please resolve the conflict..

**Comment by dailidong** [comment]
> Hi, please resolve the conflict..

I have solved this conflict file because it's the first contribution by @IamMujuziMoses

**Review by dailidong — COMMENTED** [review]
+1 
good job
